arangod/RestHandler/RestIndexHandler.cpp

```diff
@@ -199,7 +205,97 @@ RestStatus RestIndexHandler::getIndexes() {
+              auto const body = VPackBuffer<uint8_t>();
+              auto* pool =
+                  coll->vocbase().server().getFeature<NetworkFeature>().pool();
+              std::vector<Future<network::Response>> futures;
+              futures.reserve(shards->size());
+              std::string const prefix = "/_api/index/";
+              network::RequestOptions reqOpts;
+              reqOpts.param("withHidden", "true");
+              reqOpts.database = _vocbase.name();
+              // best effort. only displaying progress
+              reqOpts.timeout = network::Timeout(10.0);
+              for (auto const& shard : *shards) {
+                std::string const url =
+                    absl::StrCat(prefix, shard.first, "/", iid);
+                futures.emplace_back(network::sendRequestRetry(
+                    pool, "shard:" + shard.first, fuerte::RestVerb::Get, url,
+                    body, reqOpts));
+              }
+              for (Future<network::Response>& f : futures) {
+                network::Response const& r = f.get();
+
+                // Only best effort accounting. If something breaks here, we
+                // just ignore the output. Account for what we can and move on.
+                if (r.fail()) {
+                  LOG_TOPIC("afde4", INFO, Logger::CLUSTER)
+                      << "Communication error while fetching index data "
+                         "for collection "
+                      << coll->name() << " from " << r.destination;
+                  continue;
+                }
+                VPackSlice resSlice = r.slice();
+                if (!resSlice.isObject() ||
+                    !resSlice.get(StaticStrings::Error).isBoolean()) {
+                  LOG_TOPIC("aabe4", INFO, Logger::CLUSTER)
+                      << "Result of collecting index data for collection "
+                      << coll->name() << " from " << r.destination
+                      << " is invalid";
+                  continue;
+                }
+                if (resSlice.get(StaticStrings::Error).getBoolean()) {
+                  LOG_TOPIC("a4bea", INFO, Logger::CLUSTER)
+                      << "Failed to collect index data for collection "
+                      << coll->name() << " from " << r.destination;
+                  continue;
+                }
+                if (resSlice.get("progress").isNumber()) {
+                  progress += resSlice.get("progress").getNumber<double>();
+                } else {
+                  // Obviously, the index is already ready there.
+                  progress += 100.0;
+                  LOG_TOPIC("aeab4", DEBUG, Logger::CLUSTER)
+                      << "No progress entry on index " << iid << "  from "
+                      << r.destination << ": " << resSlice.toJson()
+                      << " index already finished.";
+                }
+              }
+              tmp.add("progress", VPackValue(progress / shards->size()));
```

Comment:
Let's add this only if `progress` is non-zero, since otherwise we will always see 0 in the cluster when `inBackground` is set to `false`.